### PR TITLE
Sort time series indices by time range in GetDataStreams API

### LINKED ISSUE #102088: Unneeded warning log emitted for tsdb data stream that have imported backing indices.
When using the modify data stream api to import a backing index to a tsdb data stream then it always gets added as first backing index. Even if the correct `index.time_series.start_time` and `index.time_series.end_time` index settings are set. 

This confuses the get data stream. It expects that the backing indices are in order of the time slice it represents. In case the get data stream api returns warnings. In this case the warning is not a sign of something being incorrect. 

Either the logic of get data stream api should be updated to not log a warning message in the case the indices are out of order. Or the modify data stream api should insert the backing index into the right slot.

### PR DESCRIPTION
**Pull request by lkts**
Previously this logic assumed that time series indices are sorted by their time series range. This is problematic because data stream APIs don't actually enforce this invariant. As seen in below issues it is possible to add an index to data stream that will break the sort order either manually or via DSL.

This PR simply sorts indices in needed order before working with them. It seems appropriate given that this API is not performance sensitive.

Closes #102088, #106890.

### CODE REVIEW

File: modules/data-streams/src/main/java/org/elasticsearch/datastreams/action/GetDataStreamsTransportAction.java [hunk 1]
```diff
@@ -151,46 +151,62 @@ static GetDataStreamAction.Response innerOperation(
...
+
+                    if (currentMergedRange == null) {
+                        currentMergedRange = new Tuple<>(start, end);
+                    } else if (currentMergedRange.v2().compareTo(start) == 0) {
+                        currentMergedRange = new Tuple<>(currentMergedRange.v1(), end);
+                    } else if (currentMergedRange.v2().compareTo(start) < 0) {
+                        mergedRanges.add(currentMergedRange);
+                        currentMergedRange = new Tuple<>(start, end);
                     } else {
                         String message = "previous backing index ["
```
Comment (martijnvg) [inline modules/data-streams/src/main/java/org/elasticsearch/datastreams/action/GetDataStreamsTransportAction.java:188]: With the indices getting sorted, I think don't we will ever log this warning? Maybe we should remove it and replace this with an `assert`?

**Review by martijnvg — COMMENTED** [review]
Looks good! I left one question.

File: modules/data-streams/src/main/java/org/elasticsearch/datastreams/action/GetDataStreamsTransportAction.java (hunk 1 above)
Comment (lkts) [reply modules/data-streams/src/main/java/org/elasticsearch/datastreams/action/GetDataStreamsTransportAction.java:188]: Yes, this is a "should never happen" thing.

**Review by martijnvg — APPROVED** [review]
lgtm

**Pull request merged**